pkg/aws/throttle/predict.go

```diff
@@ -0,0 +1,32 @@
+package throttle
+
+import (
+	"github.com/aws/aws-sdk-go/aws/request"
+	"regexp"
+)
+
+type Predict func(r *request.Request) bool
```

Comment:
What is this "predicting"? Seems like it's a matcher of sorts?

Reply:
I can also name it as "condition" is it ok to you?